src/v/iceberg/table_update_applier.cc

```diff
@@ -34,16 +34,13 @@ struct update_applying_visitor {
         }
         if (update.last_column_id.has_value()) {
             auto new_last_col_id = update.last_column_id.value();
-            if (new_last_col_id < meta.last_column_id) {
-                vlog(
-                  log.error,
-                  "Expected new last column id to be >= previous last column "
-                  "id: {} < {}",
-                  new_last_col_id,
-                  meta.last_column_id);
-                return outcome::unexpected_state;
-            }
-            meta.last_column_id = new_last_col_id;
+            // NOTE: we can drop fields now, so new_last_col_id could be
```

Comment:
nit: drop the "now" to make this comment context/history-agnostic

Reply:
done